src/home/homework-luks.c

```diff
@@ -344,6 +322,66 @@ static int luks_try_passwords(
         return -ENOKEY;
 }
 
+static int luks_get_volume_key(
+                UserRecord *h,
+                struct crypt_device *cd,
+                const PasswordCache *cache,
+                void *volume_key,
+                size_t *volume_key_size,
+                key_serial_t *ret_key_serial) {
+
+        char **list;
+        size_t vks;
+        int r;
+
+        assert(h);
+        assert(cd);
+        assert(volume_key);
+        assert(volume_key_size);
+
+        if (cache && cache->volume_key) {
+                /* Shortcut: If volume key was loaded from the keyring then just use it */
+                if (cache->volume_key_size > *volume_key_size)
+                        return log_error_errno(SYNTHETIC_ERRNO(-ENOMEM), /* Matches error from crypt_volume_key_get */
```

Comment:
(Continued from https://github.com/systemd/systemd/pull/31153#discussion_r1507971246)

Instead of an assert I made it log an error message and return -ENOMEM, like cryptsetup would in this same situation

Maybe making homework crash would be a better approach. Let me know

Reply:
sounds more like a job for ENOBUFS... 

but please drop the `-`, log_error_errno()/SYNTHETIC_ERRNO() both can deal with both negative and possitive errnos, and treat them the same, hence it's redundant

Reply:
hmm, this is not actually addressed yet

Reply:
ENOBUFS makes more sense, for sure. Cryptsetup uses ENOMEM for the same scenario, however